transport.go

```diff
@@ -196,40 +215,64 @@ func (m *MockTransport) RegisterResponder(method, url string, responder Responde
 	m.mu.Unlock()
 }
 
-// RegisterResponderWithQuery is same as RegisterResponder, but it doesn't depend on query items order
+// RegisterResponderWithQuery is same as RegisterResponder, but it
+// doesn't depend on query items order. See
+// RegisterResponderWithQueryValues to pass several values for only
+// one parameter.
 func (m *MockTransport) RegisterResponderWithQuery(method, path string, query map[string]string, responder Responder) {
-	url := path
-	mapQuery := make(map[string][]string, len(query))
+	mapQuery := make(url.Values, len(query))
 	for key, e := range query {
 		mapQuery[key] = []string{e}
 	}
-	queryString := mapToSortedQuery(mapQuery)
-	if queryString != nil {
-		url = path + "?" + *queryString
+	if queryString := sortedQuery(mapQuery); queryString != "" {
```

Comment:
At this point you could do `return RegisterResponderWithQueryValues(method, mapQuery, responder)`

Reply:
done